hw/ip/otbn/data/otbn.ld

```diff
@@ -0,0 +1,39 @@
+/* Copyright lowRISC contributors. */
+/* Licensed under the Apache License, Version 2.0, see LICENSE for details. */
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+
+  OTBN has a pure Harvard architecture, with instruction and data
+  memory both starting at address 0.
+
+  We give each 1MiB of space: too much for the eventual chip - let's
+  come back to this once we know how much space we'll have or (harder)
+  auto-generate it from the address map.
+
+  Because we have the Harvard architecture, we need to alias one of
+  the memories so that LMAs are distinct. We alias instruction memory,
+  putting a load-time version of it at 16 MiB.
+
+*/
+MEMORY
+{
+    imem (x)  : ORIGIN = 0, LENGTH = 1M
+    dmem (rw) : ORIGIN = 0, LENGTH = 1M
+
+    /* Somewhere to put LMAs for data that should end up in IMEM */
+    imem_load (rw) : ORIGIN = 16M, LENGTH = 1M
+}
+
+SECTIONS
+{
+    .text :
+    {
+        *(.text*)
+    } >imem AT >imem_load
+
+    .data :
```

Comment:
Should we add the alignments here?

Reply:
I wasn't sure what the alignment should be. Are e.g. `lw` instructions also required to be 32-byte aligned?

Reply:
For data: LW/SW memory addresses must be 32b aligned, BN.LID/BN.SID memory addresses must be 256b aligned. It should be OK to force 256b alignment for the sections themselves.

For instructions: always 32b aligned.

Reply:
Cool, I'll update to match that.

Reply:
Hmm. Actually, this isn't quite what we need. An `ALIGN` keyword on the (output) section forces alignment for that section. Since they both start at VMA 0, this doesn't actually do anything (I just tried!). The input sections might well want aligning, but that's a job for whatever produces them (probably some assembly code).

Reply:
Oh yes, so the `ALIGN` of the sections would be more for doc purposes. Feel free to add/leave out.

That still leaves the alignment for the bss/data sections copied below. We probably could force alignment on them with `. = ALIGN(32);` between the blocks just to be sure.

Reply:
Good point, and that can't hurt. I've also just re-instated the `if __name__ == "__main__"` bit at the bottom of the script - I seem to have dropped it at some point: oops!